adobe/react-spectrum
# fix RSP-1414/RSP-1415 Calendar has both grid and table - cannot use with VO

Fixes https://jira.corp.adobe.com/browse/RSP-1414 and addresses https://jira.corp.adobe.com/browse/RSP-1415.

calendarBodyProps sets `role=grid` yet `calendarBase` has a` table` as a child of this. This leads to inability to navigate the calendar `grid` using VoiceOver.

Fixed by:

Adding `role="presentation` to the `table` (`.spectrum-Calendar-table`) AND add `role=row` to each of the `tr`'s in `CalendarTableHeader` and `CalendarTableBody` AND add `role=columnheader` to the `th` in `CalendarTableHeader` AND verify that the cells in `CalendarTableBody` have `role=gridcell` on them.

Also, Calendar now marshalls focus to the focused cell in the so that the VoiceOver Control+Option+Space command will click the date, rather than the center of the calendar body grid with `aria-activedecendant`.

Regarding RSP-1415, the calendar grid cells now receive focus and expose the selected state.  Adding `button` children to each cell only adds verbosity will no added benefit.


Closes [RSP-1414](https://jira.corp.adobe.com/browse/RSP-1414) and [RSP-1415](https://jira.corp.adobe.com/browse/RSP-1415)

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [RSP-1414](https://jira.corp.adobe.com/browse/RSP-1414) and [RSP-1415](https://jira.corp.adobe.com/browse/RSP-1415).

## 📝 Test Instructions:

1. In Safari on macOS, open https://reactspectrum.blob.core.windows.net/reactspectrum/996c0552644cd665c816632e21ee7c2ad4cb87ac/iframe.html?id=rangecalendar--autofocus&providerSwitcher-locale=&providerSwitcher-theme=&providerSwitcher-scale=&providerSwitcher-toastPosition=top
2. Verify that the first selected cell, Wednesday, June 5, 2019, has received focus with `autoFocus`.
3. Use Arrow keys to move focus within the Calendar grid.
4. Hold down Arrow key and verify that focus is not lost when navigating to the previous or next month.
5. On MacOS, enable VoiceOver.
6. Use Arrow keys to navigate to the 2nd date of the month.
7. Use the Control+Option+Space VoiceOver command to activate the focused date, beginning range selection.
8. Use Control+Option+ArrowDown to navigate to the next row in the Calendar table using VoiceOver table navigation
9. Use Control+Option+Space to activate the focused date and complete the range selection.


## 🧢 Your Team:

Accessibility
